pinot-core/src/main/java/com/linkedin/pinot/core/segment/creator/impl/SegmentIndexCreationDriverImpl.java

```diff
@@ -167,7 +190,13 @@ public void buildStarTree() throws Exception {
     LOGGER.info("Start building star tree!");
     totalDocs = 0;
     while (recordReader.hasNext()) {
+      //PlainFieldExtractor conducts necessary type conversions
       GenericRow row = readNextRowSanitized();
+      //add default hll value in each row
+      //must be called after previous step since type conversion for derived values is unnecessary
+      if (enableHllIndex) {
+        populateDefaultDerivedColumnValues(row);
```

Comment:
always call populateDefaultDerivedColumnValues(). move the if condition inside method